poppy/wavefront.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import division
+import numpy as np
+import matplotlib.pyplot as plt
+import scipy.constants as const
+import astropy.units as u
+from copy import deepcopy
+from scipy.linalg import lstsq
+
+from poppy.fresnel import FresnelWavefront, QuadraticLens
+
+
+class PhysicalFresnelWavefront(FresnelWavefront):
+    """
+    This class extends the capabilities of poppy's FresnelWavefront class.
+    
+    The extension includes the redefinition of the wavefront array to
+    represent an electric field with the appropriate units V.m^-1.
+    Some utility functions have been defined to be able to compute the beam
+    radius and beam quality factor. Additionally, a wavelength scaling has 
+    been introduced to simulate a beam quality factor.
+    
+    Parameters
+    ----------
+    M2 : float
+        Beam quality factor.
+    n0 : float
+        Ambient refractive index.
+    """
+    
+    def __init__(self,
+                 beam_radius,
+                 units=u.m,
+                 rayleigh_factor=2.0,
+                 oversample=2,
+                 wavelength=1064.0e-9,
+                 npix=1024,
+                 M2=1.0,
+                 n0=1.00027398, # refractive index of air @ 15°C, lambda=1064nm
+                 **kwargs):
+        super(PhysicalFresnelWavefront, self).__init__(
+                beam_radius=beam_radius,
+                units=units,
+                rayleigh_factor=rayleigh_factor,
+                oversample=oversample,
+                wavelength=wavelength*M2/n0, # scale the wavelength and apply refractive index
+                npix=npix,
+                **kwargs
+                )
+        self.lam = wavelength/n0 # its unscaled value is needed to compute M2
+        self.n0 = n0
+        self.npix = npix*oversample
+        
+        return
```

Comment:
This and other bare return statements (not returning any value) aren't necessary.